README.md
# ARInside: Remedy Documentation Generator
ARInside generates a static set of HTML files that provide a means for navigating Remedy workflow.

## Software Information
Please see the release notes about latest changes. For changes in version 3.1.0 you can look into the What's New section of that version. To get started with ARInside please check out the Documentation page.

| Release | Platform(s)                   | Written In |
| ------- | ----------------------------- | ---------- |
| 3.1.2   | Windows, Linux, Solaris, AIX* | C++        |

* - experimental support


RELEASE_NOTES.txt
Here are the Changes between versions and the legend

N - Note
E - Enhancement
D - Defect fixed

2014-09-01
[3.1.1]
E - new filter control on the schema page workflow-tab
E - added full support for currency fields (#132)
E - advanced workflow using $SCHEMA$ is now documented using the attached schemas instead of
    the sample data form (#134)
E - added keyword support the to service action sample data
E - added support for the "Inline Form" option of open-window actions
E - audit and archive forms aren't documented as regular forms any more (#130) and could be easily
    identified on the all-forms page
E - audit log forms show references to the source form(s) in the audit section
E - added icons for different schema types and for overlay and custom objects

D - fixed a crash while running ARInside on a 5.x server (#133)
D - fixed a crash during role documentation (#131)
D - in some cases the qualifications were documented incorrectly (regarding used delimiter, e.g. 
    'field' or $field$)
D - fixed a crash in the directory detection of program's executable

N - on the schema page, actlink- and filter-guides aren't shown on the reference tab any more. They
    are listed on the workflow tab. 

2014-03-30
[3.1.0]
E - Redesign of the whole schema-page (added tab-control to organize the different informations)
E - added database view-names and schemaid to the schema-page (#58, #59)
E - added support for 8.0 and 8.1 keywords and functions
E - added a new filter-control to the field list on the schema-page 
E - moved information which were stored in different pages directly into the schema-page 
    (removed the additional navigation items on schema-pages)
E - added accordion control to the general-tab of the schema-page holding all the different details
E - added default vui information to the schema page
E - added form tag support (8.0 feature)
E - schema references are now distinguished between data read, data writen
E - added service-calls references to the schema page (#97)
E - added full support for filterApi calls (input-mapping, output-mapping and references are created)
E - set-field actions show "Data Source" like in devstudio
E - added filtering support to all the overview/list-pages (actlinks, filters, escalations, menus,
    alguides, fltguides, applications, packing-lists, groups, roles, users, images); filtering is
    limited to 100 results and next chunk of results could be fetched by user
E - added tar/gzip support and moved the static-resources out of the executable
E - delete-workflow is listed on the schema page (see references tab)
E - references between containers are now shown, so you could see if a actlink-guide, filter-guide
    application or packing list is part of another packing list (#16 & #98)

D - fixed an error with setfields-all-matching-ids in case it is used in combination with sample-data
D - fixed an error in sample data resolving, if the sample data is read from current server (uses @)
D - fixed a crash if webservice was attached to a missing form (or form is blacklisted)
D - push-field actions with all-matching-id mapping, the source and target forms were switched

2012-06-06
[3.0.3]
E - Support for overlay feature (#102)
E - New ini-setting to disable overlay support
E - Improved support for 7.6.04 servers (keywords, display-props, server-infos)
E - Added again slow object loading as a fallback
E - New --slow command line argument to disable fast object loading completely
E - keywords used in default-values of fields are documented now (#116)
E - New page showing all notification texts
E - New page showing overlay- and custom-objects

D - in notification actions first field from "select-field-list" wasn't documented (#113)
D - objectOwner in a webservice was not linking to the forms page (#115)
D - some value assignments to view-fields were not documented at all
D - object-list sorting is case-insensitiv again (#105)
D - function-value assigned to enum field could lead to crash (#108)
D - references to fields in OpenWindow's Close-Mapping show "Else-Action 1048" (#109)
D - "PushFields to *this* form" section is missing escalations (#103)


2011-03-01
[3.0.2]
E - Apache/zlib-Support added to generate compressed documentation
E - improved readability of assignments (e.g. push fields, set fields)
E - added container references to menus (#69)
E - moved form-subpage-navigation from right side to the left (#73)
E - visuals of letter-indexes improved (link only to pages with at least one object)
E - now field overview lists show more details for join, view and vendor forms
E - field-detail-pages of join, view and vendor forms show details about its base field/form
E - 7.6.03 and 7.6.04 support added (not finished yet)
E - support for table-columns of type 'Display' and 'Data' added
E - field-detail-pages for table fields lists table-columns in a sortable column
E - added support for more display properties and their values (7.5 / 7.6.03 / 7.6.04)
E - new APITimeout option to avoid timeout-error during object-loading (#96)
E - Support for field-patterns added
E - complete redesign of the internal object referencing
E - lots of performance improvements
E - added webservice-references to schema-pages
E - The Escalations-ByAction-page supports service action
E - added page that lists all error handlers (#95)
E - added separate "Attached Workflow" section to fields (e.g. to show workflow attached to buttons)
E - removed duplicate workflow-references (e.g. if a field is used in multiple setfield-actions in the same workflow object)
E - workflow-references table of a global field show workflow of other schemas with the same global-field-id now
E - support for 64-bit platforms added
E - experimental support for AIX platform added

D - join member forms doesn't link to the join form (#91)
D - column-field references showed wrong source table (#92)

resource.h
//{{NO_DEPENDENCIES}}
// Microsoft Visual C++ generated include file.
// Used by ARInside.rc
//
#define ID_ACTIVE_LINK                  101
#define ID_AL_GUIDE                     102
#define ID_APPLICATION                  103
#define ID_ARS_EDIT                     104
#define ID_CHAPTER                      105
#define ID_DISABLED                     106
#define ID_DOC                          107
#define ID_DOWN                         108
#define ID_EDIT                         109
#define ID_EMPTY_LIST                   110
#define ID_ESCALATION                   111
#define ID_FILTER                       112
#define ID_FILTER_GUIDE                 113
#define ID_FOLDER                       114
#define ID_GROUP                        115
#define ID_HIDDEN                       116
#define ID_IMAGE                        117
#define ID_INFO                         118
#define ID_MENU                         119
#define ID_NEXT                         120
#define ID_PACKING_LIST                 121
#define ID_PREV                         122
#define ID_SCHEMA                       123
#define ID_SCRIPT                       124
#define ID_SEARCH                       125
#define ID_SERVER                       126
#define ID_SORT_ASC                     127
#define ID_SORT_DESC                    128
#define ID_SORTSCRIPT                   129
#define ID_STYLE_CSS                    130
#define ID_TABSCRIPT                    131
#define ID_UP                           132
#define ID_USER                         133
#define ID_VISIBLE                      134
#define ID_WEBSERVICE                   135
#define ID_SCHEMAPAGE_JS                136
#define ID_JQUERY_JS                    137
#define ID_JQUERY_UI_JS                 138
#define ID_JQUERY_UI_CUSTOM_CSS         139
#define ID_JQUERY_ADDRESS_JS            140
#define ID_JQUERY_TIMER_JS              141
#define ID_JQUERY_UI_IMAGE_BGGLASS_E6   142
#define ID_JQUERY_UI_IMAGE_BGGLASS_FF   143

// Next default values for new objects
// 
#ifdef APSTUDIO_INVOKED
#ifndef APSTUDIO_READONLY_SYMBOLS
#define _APS_NEXT_RESOURCE_VALUE        136
#define _APS_NEXT_COMMAND_VALUE         40001
#define _APS_NEXT_CONTROL_VALUE         1001
#define _APS_NEXT_SYMED_VALUE           101
#endif
#endif
